sunpy/data/_sample.py

```diff
@@ -40,24 +39,67 @@
     "EIT_195_IMAGE": "eit_l1_20110607_203753.fits",
     "RHESSI_IMAGE": "hsi_image_20110607_063300.fits",
     "CALLISTO_SPECTRUM": "BIR_20110607_062400_10.fit",
-    # Not in the sample-data repo
-    # "RHESSI_EVENT_LIST": "hsi_calib_ev_20020220_1106_20020220_1106_25_40.fits",
     "SWAP_LEVEL1_IMAGE": "swap_lv1_20110607_063329.fits",
     "EVE_TIMESERIES": "20110607_EVE_L0CS_DIODES_1m.txt",
-    # Uncomment this if it needs to be used. Commented out to save bandwidth.
-    # "LYRA_LIGHTCURVE": ("lyra_20110810-000000_lev2_std.fits.gz", ,
     "LYRA_LEVEL3_TIMESERIES": "lyra_20110607-000000_lev3_std.fits",
     "GOES_XRS_TIMESERIES": "go1520110607.fits",
     "GBM_TIMESERIES": "glg_cspec_n5_110607_v00.pha",
     "RHESSI_TIMESERIES": "hsi_obssumm_20110607_025.fits",
-    "NORH_TIMESERIES": "tca110607.fits"
+    "NORH_TIMESERIES": "tca110607.fits",
 }
 
 # Reverse the dict because we want to use it backwards, but it is nicer to
 # write the other way around
-_sample_files = {v: k for k, v in _sample_data.items()}
+_SAMPLE_FILES = {v: k for k, v in _SAMPLE_DATA.items()}
+
+
+def _download_sample_data(base_url, sample_files, overwrite):
+    """
+    Downloads a list of files.
 
-_error = namedtuple("error", ("filepath_partial", "url", "response"))
+    Parameters
+    ----------
+    base_url : str
+        Base URL for each file.
+    sample_files : list of tuples
+        List of tuples that are (URL_NAME, SAVE_NAME).
+    overwrite : bool
+        Will overwrite a file on disk if True.
+
+    Returns
+    -------
+    `parfive.Results`
+        Download results. Will behave like a list of files.
+    """
+    dl = Downloader(overwrite=overwrite, progress=False)
+    for url_file_name, fname in sample_files:
+        url = urljoin(base_url, url_file_name)
+        dl.enqueue_file(url, filename=fname)
+    results = dl.download()
+    return results
+
+
+def _retry_sample_data(results):
+    # In case we have a broken file on disk, overwrite it.
+    dl = Downloader(overwrite=True, progress=False)
+    for err in results.errors:
+        file_name = err.filepath_partial().name
+        log.debug(
+            f"Failed to download {_SAMPLE_FILES[file_name]} from {err.url}: {err.exception}")
+        # Update the url to a mirror and requeue the file.
+        new_url = urljoin(_BASE_URLS[1], file_name)
```

Comment:
Probably doesn't really matter, but this now limits us to two mirrors?

Reply:
It does but my head hurt trying to work out the logic for a loop. I can try to figure it out if you prefer.

Reply:
nah, we can cross that bridge when we come to it.